backend/src/api/routes/rooms.ts

```diff
@@ -76,4 +76,34 @@ router.put(
   },
 );
 
+router.put('/:rooms/update', permissionMiddleware(PermissionLevel.admin), async (req, res) => {
+  const { friendlyName, capacity, needAccess, description, roomRules, requestLimit } = req.body;
+  if (
+    (capacity !== undefined && isNaN(parseInt(capacity))) ||
+    capacity === null ||
+    capacity < 0 ||
+    (needAccess !== undefined && typeof needAccess !== 'boolean') ||
```

Comment:
why are both checks needed? if `needAccess` is undefined then its type is not boolean

Reply:
true true

Reply:
lowkey if you need more of these it could be beneficial to install an abstraction library like [yup](https://www.npmjs.com/package/yup)

Reply:
Yes, maybe eventually we'll refactor. Don't have the time to mess with it right now